### PR TITLE
Add stablehlo-lsp-server

### PR DESCRIPTION
**Pull request by joker-eph**
This allows VSCode plugin integration as documented here:

https://mlir.llvm.org/docs/Tools/MLIRLSP/#mlir-lsp-language-server--mlir-lsp-server

### CODE REVIEW

File: stablehlo/tools/StableHloLspServerMain.cpp [hunk 1]
```diff
@@ -0,0 +1,22 @@
+#include "mlir/Tools/mlir-lsp-server/MlirLspServerMain.h"
```
Comment (burmako) [inline stablehlo/tools/StableHloLspServerMain.cpp:1]: Nit: Let's rename to `StablehloLspServerMain.cpp` to follow the conventional spelling of StableHLO in code: "Stablehlo" (without capitalization except for the first letter as appropriate).

**Review by burmako — APPROVED** [review]
Thank you! Looks great to me, except for one stylistic comment.

**Comment by burmako** [comment]
Oops right I clicked "Merge", I realized that we'll also need to update the Bazel build. Will do that in a moment.

**Comment by joker-eph** [comment]
Hopefully that didn't break anything? (there is no test relying on this, it just means the lsp-server can't be built with Bazel ATM)

**Comment by burmako** [comment]
Yeah, everything's alright - although I've added stablehlo-lsp-server to both Bazel and CMake CI to make sure that it stays this way :)

**Pull request merged**